src/Microsoft.IdentityModel.Xml/DsigSerializer.cs

```diff
@@ -174,13 +135,176 @@ public virtual KeyInfo ReadKeyInfo(XmlReader reader)
             return keyInfo;
         }
 
+
+        /// <summary>
+        /// Creates a new <see cref="KeyInfo"/> object.
+        /// </summary>
+        /// <param name="reader">The current <see cref="XmlReader"/>.</param>
+        protected virtual KeyInfo CreateKeyInfo(XmlReader reader)
+        {
+            XmlUtil.CheckReaderOnEntry(reader, XmlSignatureConstants.Elements.KeyInfo, XmlSignatureConstants.Namespace);
```

Comment:
This is going to get called twice now.

Reply:
Actually, when removing this check, I get a warning.

> CA1062: In externally visible method 'KeyInfo DSigSerializer.CreateKeyInfo(XmlReader reader)', validate parameter 'reader' is non-null before using it. If appropriate, throw an ArgumentNullException when the argument is null or add a Code Contract precondition asserting non-null argument.

That's probably why I had it in there.

Reply:
Yeah, 3 years is too long.
Let's leave it for now.